test/integration/server_stats.h

```diff
@@ -38,6 +38,14 @@ class IntegrationTestServerStats {
    */
   virtual void waitForCounterExists(const std::string& name) PURE;
 
+  /**
+   * Wait until a histogram has samples.
+   * @param name histogram name.
+   */
+  virtual void waitUntilHistogramHasSamples(
+      const std::string& name,
+      std::chrono::milliseconds timeout = std::chrono::milliseconds::zero()) PURE;
```

Comment:
I'd prefer waiting for Default timeout and passing on the status-failure if the histogram doesn't exist so 
ASSERT_TRUE(waitUntilHistogram...)?

Reply:
Hmm, everything else in that file follows this pattern tho and the default would still be zero/unbound.

Reply:
Yeah, I'm trying to eradicate infinite timeouts in Envoy tests.  I can refactor this with the rest though.